x-pack/plugins/security_solution/public/actions/add_to_timeline/data_provider.ts

```diff
@@ -0,0 +1,122 @@
+  queryMatch: {
+    field,
+    value,
+    operator: IS_OPERATOR,
+  },
+});
+
+export interface CreateDataProviderParams {
+  contextId?: string;
+  eventId?: string;
+  field?: string;
+  fieldFormat?: string;
+  fieldFromBrowserField?: BrowserField;
+  fieldType?: string;
+  isObjectArray?: boolean;
+  linkValue?: string | null;
+  values: string | string[] | null | undefined;
+}
+
+export const createDataProviders = ({
+  contextId,
+  eventId,
+  field,
+  fieldFormat,
+  fieldType,
+  linkValue,
+  values,
+}: CreateDataProviderParams) => {
+  if (field == null || values === null || values === undefined) return null;
+  const arrayValues = Array.isArray(values) ? values : [values];
+  return arrayValues.reduce<DataProvider[]>((dataProviders, value, index) => {
+    let id: string = '';
+    const appendedUniqueId = `${contextId}-${eventId}-${field}-${index}-${value}`;
+
+    if (fieldType === GEO_FIELD_TYPE || field === MESSAGE_FIELD_NAME) {
+      return dataProviders;
+    } else if (fieldType === IP_FIELD_TYPE) {
+      id = `formatted-ip-data-provider-${contextId}-${field}-${value}-${eventId}`;
+      if (isString(value) && !isEmpty(value)) {
+        let addresses = value;
+        try {
+          addresses = JSON.parse(value);
+        } catch (_) {
+          // Default to keeping the existing string value
+        }
+        if (isArray(addresses)) {
+          addresses.forEach((ip) => dataProviders.push(getDataProvider(field, id, ip)));
+        } else {
+          dataProviders.push(getDataProvider(field, id, addresses));
+        }
+        return dataProviders;
+      }
+    } else if (PORT_NAMES.some((portName) => field === portName)) {
+      id = `port-default-draggable-${appendedUniqueId}`;
+    } else if (field === EVENT_DURATION_FIELD_NAME) {
+      id = `duration-default-draggable-${appendedUniqueId}`;
+    } else if (field === HOST_NAME_FIELD_NAME) {
+      id = `event-details-value-default-draggable-${appendedUniqueId}`;
+    } else if (fieldFormat === BYTES_FORMAT) {
+      id = `bytes-default-draggable-${appendedUniqueId}`;
+    } else if (field === SIGNAL_RULE_NAME_FIELD_NAME) {
+      id = `event-details-value-default-draggable-${appendedUniqueId}-${linkValue}`;
+    } else if (field === EVENT_MODULE_FIELD_NAME) {
+      id = `event-details-value-default-draggable-${appendedUniqueId}-${value}`;
+    } else if (field === SIGNAL_STATUS_FIELD_NAME) {
+      id = `alert-details-value-default-draggable-${appendedUniqueId}`;
+    } else if (field === AGENT_STATUS_FIELD_NAME) {
+      id = `event-details-value-default-draggable-${appendedUniqueId}`;
+    } else if (
+      [
+        RULE_REFERENCE_FIELD_NAME,
+        REFERENCE_URL_FIELD_NAME,
+        EVENT_URL_FIELD_NAME,
+        INDICATOR_REFERENCE,
+      ].includes(field)
+    ) {
+      id = `event-details-value-default-draggable-${appendedUniqueId}-${value}`;
+    } else {
+      id = `event-details-value-default-draggable-${appendedUniqueId}`;
```

Comment:
in this entire if/else block, this is the only condition checked by the tests (i believe). please add tests for others